src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py

```diff
@@ -455,43 +457,125 @@ def create_resource(properties,
     return res.create_resource(properties, location, is_full_object)
 
 
-def show_resource(resource_group_name=None,
+def show_resource(resource_ids=None, resource_group_name=None,
                   resource_provider_namespace=None, parent_resource_path=None, resource_type=None,
                   resource_name=None, api_version=None):
+    if resource_ids:
+        results = []
+        for rid in resource_ids:
+            parsed_id = parse_resource_id(rid)
+            if len(parsed_id) == 1:
+                logger.error('az resource show: error: argument --ids: invalid ResourceId value: \'%s\'' % rid)
+                return
+            results.append(show_resource(
+                resource_group_name=parsed_id['resource_group'],
+                resource_provider_namespace=parsed_id['resource_namespace'],
+                parent_resource_path=parsed_id['resource_parent'],
+                resource_type=parsed_id['resource_type'],
+                resource_name=parsed_id['resource_name'], api_version=api_version))
+        return results[0] if len(results) == 1 else results
+
     res = _ResourceUtils(resource_group_name, resource_provider_namespace,
                          parent_resource_path, resource_type, resource_name,
                          None, api_version)
     return res.get_resource()
 
 
-def delete_resource(resource_group_name=None,
+def delete_resource(resource_ids=None, resource_group_name=None,
                     resource_provider_namespace=None, parent_resource_path=None, resource_type=None,
-                    resource_name=None, resource_id=None, api_version=None):
-    res = _ResourceUtils(resource_group_name, resource_provider_namespace,
-                         parent_resource_path, resource_type, resource_name,
-                         resource_id, api_version)
-    return res.delete()
-
-
-def update_resource(parameters,
+                    resource_name=None, api_version=None):
+    if not resource_ids:
+        res = _ResourceUtils(resource_group_name, resource_provider_namespace,
+                             parent_resource_path, resource_type, resource_name, None, api_version)
+        return res.delete()
+
+    from collections import deque
+    parsed_ids = deque()
+    for rid in resource_ids:
+        parsed_id = parse_resource_id(rid)
+        if len(parsed_id) == 1:
+            logger.error('az resource delete: error: argument --ids: invalid ResourceId value: \'%s\'' % rid)
+            return
+        parsed_ids.append(parsed_id)
+
+    threads = deque()
+    results = []
+    deleted = True
+    while parsed_ids and deleted:
+        deleted = False
+        for _ in range(len(parsed_ids)):
+            parsed_id = parsed_ids.pop()
+            try:
+                threads.append(delete_resource(
```

Comment:
Unless the `delete_resource` is an async operation this is not run in parallel. The name "threads" is overkill and the usage of the deque data structure here is over-engineered. A single loop should be enough.

Reply:
After discussion with Willie, we cleared this up.